bsp/raspberrypi/rp2040/src/hal/spi.zig

```diff
@@ -12,57 +12,141 @@ const hw = @import("hw.zig");
 
 const SpiRegs = microzig.chip.types.peripherals.SPI0;
 
+pub const Motorola = struct {
+    pub const Polarity = enum(u1) {
+        low = 0,
+        high = 1,
+    };
+
+    pub const Phase = enum(u1) {
+        first_edge = 0,
+        second_edge = 1,
+    };
+
+    /// Low means clock is steady state low, high means steady state high
+    clock_polarity: Polarity = .low,
+
+    /// Controls whether data is captured on the first or second clock edge transition
+    clock_phase: Phase = .first_edge,
+};
+
+pub const FrameFormat = union(enum) {
+    motorola: Motorola,
+    texas_instruments,
+    ns_microwire,
+};
+
 pub const Config = struct {
     clock_config: clocks.GlobalConfiguration,
-    tx_pin: ?gpio.Pin = gpio.num(19),
-    rx_pin: ?gpio.Pin = gpio.num(16),
-    sck_pin: ?gpio.Pin = gpio.num(18),
-    csn_pin: ?gpio.Pin = gpio.num(17),
-    baud_rate: u32 = 1000 * 1000,
+    /// Number of bits in a frame can't be less than 4 or more than 16
+    data_width: u4 = 8,
```

Comment:
Use u5 here, as u4 cannot encode `16`

Reply:
Lol great catch, I think I'll just bite the bullet and make this an `enum(u5)`, it's not that many values and it makes it more explicit.